crates/nu-command/src/viewers/table.rs

```diff
@@ -523,48 +540,53 @@ fn handle_row_stream(
         _ => stream,
     };
 
-    let head = call.head;
-    let width_param: Option<i64> = call.get_flag(engine_state, stack, "width")?;
+    if let Some(limit) = cfg.abbreviation {
+        // todo: could be optimized cause we already consumed the list there's no point in goint back to pagination;
 
-    let collapse: bool = call.has_flag("collapse");
+        let mut data = consume_stream(stream, &ctrlc);
```

Comment:
`table` shouldn't collect its input, as the input could be potentially infinite and would cause nothing to be printed to the screen.

Instead of show the first few rows and the last few, we should show the first x rows and then tell the user that there more rows that have been abbreviated.